NPE during textDocument/definition

I was not able to reproduce it, but at some point I ctrl+clicked on my application.properties and saw an error popup n VS Code. Stacktrace shows:

```
SEVERE: Internal error: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.commons.MicroProfileProjectInfo.getProperties()" because "info" is null
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.commons.MicroProfileProjectInfo.getProperties()" because "info" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.commons.MicroProfileProjectInfo.getProperties()" because "info" is null
	at org.eclipse.lsp4mp.utils.PropertiesFileUtils.getProperty(PropertiesFileUtils.java:184)
	at org.eclipse.lsp4mp.services.properties.PropertiesFileDefinition.findDefinition(PropertiesFileDefinition.java:98)
	at org.eclipse.lsp4mp.services.properties.PropertiesFileLanguageService.findDefinition(PropertiesFileLanguageService.java:156)
	at org.eclipse.lsp4mp.ls.properties.PropertiesFileTextDocumentService.lambda$definition$7(PropertiesFileTextDocumentService.java:196)
	at org.eclipse.lsp4mp.ls.commons.ModelTextDocuments.lambda$computeModelAsyncCompose$1(ModelTextDocuments.java:146)
	... 7 more

[Error - 18:28:21] Request textDocument/definition failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.commons.MicroProfileProjectInfo.getProperties()" because "info" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.commons.MicroProfileProjectInfo.getProperties()" because "info" is null
	at org.eclipse.lsp4mp.utils.PropertiesFileUtils.getProperty(PropertiesFileUtils.java:184)
	at org.eclipse.lsp4mp.services.properties.PropertiesFileDefinition.findDefinition(PropertiesFileDefinition.java:98)
	at org.eclipse.lsp4mp.services.properties.PropertiesFileLanguageService.findDefinition(PropertiesFileLanguageService.java:156)
	at org.eclipse.lsp4mp.ls.properties.PropertiesFileTextDocumentService.lambda$definition$7(PropertiesFileTextDocumentService.java:196)
	at org.eclipse.lsp4mp.ls.commons.ModelTextDocuments.lambda$computeModelAsyncCompose$1(ModelTextDocuments.java:146)
	... 7 more

```

This is using the latest vscode-microprofile pre-release build (v0.7.2023031404)

--- a/microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/properties/PropertiesFileDefinition.java
+++ b/microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/properties/PropertiesFileDefinition.java
@@ -87,6 +87,12 @@ public class PropertiesFileDefinition {
 				}
 			}
 
+			// Here the MicroProfile project information is required. 
+			if (projectInfo == null) {
+				// The project is not loaded, return an empty definition. 
+				return getEmptyDefinition();
+			}
+			
 			// Get the property at the given position
 			PropertyKey key = getPropertyKey(node);
 

--- a/microprofile.ls/org.eclipse.lsp4mp.ls/src/test/java/org/eclipse/lsp4mp/services/properties/PropertiesFileDefinitionTest.java
+++ b/microprofile.ls/org.eclipse.lsp4mp.ls/src/test/java/org/eclipse/lsp4mp/services/properties/PropertiesFileDefinitionTest.java
@@ -15,6 +15,7 @@ import static org.eclipse.lsp4mp.services.properties.PropertiesFileAssert.testDe
 
 import java.util.concurrent.ExecutionException;
 
+import org.eclipse.lsp4mp.commons.MicroProfileProjectInfo;
 import org.eclipse.lsp4mp.ls.commons.BadLocationException;
 import org.junit.Test;
 
@@ -127,6 +128,12 @@ public class PropertiesFileDefinitionTest {
 		testDefinitionFor(value);
 		value = "quarkus.datasource.driver=XXXX|";
 		testDefinitionFor(value);
+	}
+
+	@Test
+	public void noProject() throws BadLocationException, InterruptedException, ExecutionException {
+		String value = "quarkus.datasour|ce.driver=XXXX";
+		testDefinitionFor(value, null, (MicroProfileProjectInfo) null, null);
 	}
 
 }

--- a/microprofile.ls/org.eclipse.lsp4mp.ls/src/test/java/org/eclipse/lsp4mp/services/properties/expressions/PropertiesFileExpressionDefinitionTest.java
+++ b/microprofile.ls/org.eclipse.lsp4mp.ls/src/test/java/org/eclipse/lsp4mp/services/properties/expressions/PropertiesFileExpressionDefinitionTest.java
@@ -141,4 +141,16 @@ public class PropertiesFileExpressionDefinitionTest {
 				"test.property=salutations";
 		testDefinitionFor(value, PROPERTY_DOCUMENT_NAME, ll(PROPERTY_DOCUMENT_NAME, r(0, 19, 37), r(1, 0, 18)));
 	}
+
+	@Test
+	public void noProject() throws BadLocationException, InterruptedException, ExecutionException {
+		// Same test than below but with no project.
+		String value = "other.property = ${%dev.test.prop|erty}\n" + //
+				"%dev.test.property=hi\n" + //
+				"%prop.test.property=hello\n" + //
+				"test.property=salutations";
+		testDefinitionFor(value, PROPERTY_DOCUMENT_NAME, (MicroProfileProjectInfo) null, null,
+				ll(PROPERTY_DOCUMENT_NAME, r(0, 19, 37), r(1, 0, 18)));
+	}
+
 }

